### PR TITLE
Feature/add form

### PR DESCRIPTION
**Pull request by leloufadel**
[Project requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/forms_project_v1_1.md#project-requirements)
Create a method called current_user in your ApplicationController to make current user data available to all controllers.
It will return the first user from the database.
Don't worry about proper authentication at this time. You will add it in the upcoming projects.
Create forms to perform the following functions:
Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).
Allow Users to add likes to Posts.

### CODE REVIEW

**Review by IngDavidTM — CHANGES_REQUESTED** [review]
## Hi @leloufadel 🙋‍♂️, 

Good job so far!
You have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!



## Top highlight 🎯
- The Pull Request follows the GitFlow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 

## Suggested changes

- [ ] While you did a fantastic job, as a project requirement the user should be able to add a like to posts. So kindly add the likes controller with all the necessary functionality 👍

- [x] _Check the comments under the review._ ⬇

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/show.html.erb [hunk 1]
```diff
@@ -1,25 +1,37 @@
...
-<% else %>
-  <p>No comments yet.</p>
-<% end %>
-</div>
\ No newline at end of file
+</div>
+
+<a href="<%= new_user_post_comment_path(@user, @post) %>">
+  <button id="back-btn">Add a comment</button>
+</a>
```
Comment (rloterh) [inline app/views/posts/show.html.erb]: - [ ] Although post can be created successfully for given user, I'm unable to like nor comment on a post. When I click on a view post so as to be directed to the page where I can like or comment on the post, an error is thrown like below. It appears a bug was introduced in your code in an attempt to resolve the previous issue. Kindly ensure to fix the same.

image

image

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Lala, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🏁 🚦 🌴  🏃 🌳 🏡 🏘️ 🌲 

## Highlights 👍  
- Good job so far improving the forms!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/show.html.erb (hunk 1 above)
Comment (Abenezer-Tilahun) [reply app/views/posts/show.html.erb]: - [ ] Good Job @leloufadel  Fixed as shown below  screenshot 
test

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @leloufadel 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.


## Optional suggestions 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Pull request merged**